Habitica/src/main/java/com/habitrpg/android/habitica/ui/adapter/inventory/StableRecyclerAdapter.kt

```diff
@@ -34,13 +39,14 @@ class StableRecyclerAdapter : androidx.recyclerview.widget.RecyclerView.Adapter<
 
     override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): androidx.recyclerview.widget.RecyclerView.ViewHolder =
             if (viewType == 0) {
-                val view = LayoutInflater.from(parent.context).inflate(R.layout.customization_section_header, parent, false)
-                SectionViewHolder(view)
+                val view = parent.inflate(R.layout.shop_header)
+                ShopRecyclerAdapter.ShopHeaderViewHolder(view)

```

Comment:
Unless I'm looking at the change wrong, I think this replaces all the section headers with the NPC banner? What should happen instead is that the section headers stay the same and there is a new view holder at the top of the recyclerview that has the NPC banner.

To achieve that, I would change the `getItemCount` method on the adapter, so that it returns `itemList.size + 1`, since we want that extra item for the header. And then for `getItemViewType` it would have to return a new type to distinguish the header viewholder. And places where you access the itemList would need to be offset by one, to account for the header being at the top. I hope that explanation is somewhat clear!